docs/source/reference/extra-handlers.md

```diff
@@ -0,0 +1,54 @@
+# Adding extra handlers to the hub
+
+As an alternative to [Services](./services.md), it is possible to add extra tornado handlers to the hub.
+
+They can be used to show custom pages, expose additional API endpoints and add extra functionalities to the hub.
+
+## Adding the handler to the JupyterHub config
+
+Additional handlers can be created and defined in the `jupyterhub_config.py` file,
+by extending [`c.JupyterHub.extra_handlers`](../api/app.html#jupyterhub.app.JupyterHub.extra_handlers).
+
+For example:
+
+```python
+from jupyterhub.handlers.base import BaseHandler
+from tornado.web import authenticated
+
+class CustomHandler(BaseHandler):
+    @authenticated
+    def get(self):
+        self.write(self.render_template("page.html"))
+
+c.JupyterHub.extra_handlers.extend([
+    (r"custom", CustomHandler),
+])
+```
+
+Issuing a GET request to `/hub/custom` will now render the default JupyterHub page with an empty content.
+
+## Adding a new link to the default navigation bar
+
+The new handler can be added to the navigation bar by creating a new `page.html` file template with the following content:
```

Comment:
`page.html` is perhaps a confusing name, since there's an existing `page.html` in the base templates. Maybe use `custompage.html`?

Reply:
Shouldn't it be called `page.html` to override the default navbar and include the extra item?

Reply:
To override the default navbar of the home page, we need to override `page.html` indeed :+1: 

What was a bit confusing to me was that the CustomHandler is rendering `page.html`. I believe it would be nice to have CustomHandler render a custom page instead (maybe named `custompage.html`) that extends `page.html`.
That way, in addition to the instructions on how to add an extra handler and how to override a default template to include that handler in the home page, there will also be info on how to extend a base template.

What do you think?